rag/advanced_rag/knowlege_compile/_common.py

```
@@ -673,16 +679,28 @@ def _root(i: int) -> int:
+    for group_indices in groups.values():
+        for left_start in range(0, len(group_indices), block_size):
+            left_indices = group_indices[left_start : left_start + block_size]
+            left_vectors = matrix[left_indices]
+            for right_start in range(left_start, len(group_indices), block_size):
+                right_indices = group_indices[right_start : right_start + block_size]
+                sims = left_vectors @ matrix[right_indices].T
+                if right_start == left_start:
+                    rows, cols = np.triu_indices(len(left_indices), k=1)
+                else:
+                    rows, cols = np.nonzero(sims >= ambiguous_low)
+                for row, col in zip(rows.tolist(), cols.tolist()):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)